### ISSUE TITLE
More Vanilla Turrets - incorrect or unlimited turret ranges

### ISSUE THREAD

**Issue by Kubomius** [AthenaSulisMinerva/CombatExtendedFastTrack#89]
I have a bug with More Vanilla turrets where the Gun complex, military grade turret (manned/unmanned) and precision turret gets unlimited range. I tested taking combat extended off the modlist and then it worked, so the problem is combat extended. Also the autocannon turret and uranium slug turret does not even show up as buildables (with godmode on).

**Comment by Kubomius** [comment]
Also the Vulcan turrets does not have their minimum range. I also tested it with just combat extended and more vanilla turrets and the same issues where there.

**Comment by Kubomius** [comment]
The range issue also seem to be on all of the Defensive Machine Gun Turret Pack turrets aswell

**Comment by Kubomius** [comment]
And I also just noticed that the range shown when "planting" the turret is increased substantially as soon as the turret is constructed

**Comment by sumghai** [comment]
Could you please start a separate issue for Defensive Machine Gun Turret Pack? Let's keep this discussion focused on problems with More Vanilla Turrets.

**Comment by AthenaSulisMinerva** [comment]
@sumghai - let's push this back for a later update, since we're pretty behind schedule already.

**Comment by sumghai** [comment]
No worries, makes sense.

**Comment by sumghai** [comment]
Okay @Kubomius, I think I figured out the first part of your long* list of problems.

Basically, every turret has the following values:
* `range` - the actual range the turret gun will fire in
* `specialDisplayRadius` - a circle of a given radius drawn with the building at its center; as the name implies, it is for display purposes only

What has happened is that:
* Turret gun `range` values are higher in CE than in vanilla; this is because we calculate and re-balance stats based on real-world data 
* However, we basically forgot to also update the `specialDisplayRadius` value

So basically, the turrets in CE **_do not_**, in fact, have infinite range - rather, it shows the old `specialDisplayRadius` value while it can now fire at the new (higher) `range` value. I can simply modify the patch so that the `specialDisplayRadius` will show the same value as the actual turret gun `range`.

*Again, this is why users should post each problem as a separate issue - we deal with one issue, close the ticket and move on to the next problem, instead of worrying we accidentally passed over a different problem buried in the OP.

**Comment by sumghai** [comment]
Another thing that I found:

CE will hard cap the display radius to less than 90 cells (≈1060 meters in real life), and anything above that simply won't get drawn.

Since CE generally specifies typical turret ranges of 800m (75 cells), this means I will actually have to nerf the range of some of these turrets, just so that I can draw the display radii.

**Comment by sumghai** [comment]
Fixed and ready to be rolled into the next release.

**Comment by sumghai** [comment]
Reopening because there were other issues in the chain of OPs.

**Comment by sumghai** [comment]
Minimum ranges for all applicable turrets have now been added back in.

Finally, to answer your last quesiton, @Kubomius:

>Also the autocannon turret and uranium slug turret does not even show up as buildables (with godmode on).

The original upstream CE mod (and CE:FT) removes the Autocannon and Uranium Slug (sniper) turrets, because they don't fit into CE's playstyle, and CE without other third-party mods already provides a better selection of turrets:

[Patches\Core\ThingDefs_Buildings\Buildings_Security.xml](https://github.com/AthenaSulisMinerva/CombatExtendedFastTrack/blob/6d42bf98747c29b40f09a936815ac32a2931e2bb/Patches/Core/ThingDefs_Buildings/Buildings_Security.xml#L130):
```
<!-- ========== Sniper turret ========== -->

 <Operation Class="PatchOperationRemove">
	<xpath>Defs/ThingDef[defName="Turret_Sniper"]</xpath>
</Operation>

<!-- ========== Autocannon turret ========== -->

<Operation Class="PatchOperationRemove">
	<xpath>Defs/ThingDef[defName="Turret_Autocannon"]</xpath>
</Operation>
```

I think we can call this done now (for reals).
